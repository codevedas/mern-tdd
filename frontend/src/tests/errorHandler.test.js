import React from 'react'
import ErrorHandler from '../components/errorHandler'
import errorMock from '../__mocks__/error.json'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { mount, shallow } from 'enzyme'
Enzyme.configure({ adapter: new Adapter()})
describe('Error', () => {
  it('renders without crashing', () => {
    const component = mount(<ErrorHandler/>);
    expect(component).toMatchSnapshot();
  });
  it('empty without error', () => {
    const component = shallow(<ErrorHandler/>);
    expect(component.find('#error').text()).toBe("");
  });
  it('displays with error', async () => {
    const component = mount(<ErrorHandler
      error={errorMock}
      />);
      await component.update()
    expect(component.find('#error').text()).toBe("Error: error text from json mock");
  });
});
