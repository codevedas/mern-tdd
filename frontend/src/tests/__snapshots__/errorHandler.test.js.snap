// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders without crashing 1`] = `ReactWrapper {}`;
